catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/MatchAnyValidator.java

```diff
@@ -37,42 +39,59 @@ public MatchAnyValidator(List<AttributeValidator> validators) {
-        validators
-            .stream()
-            .map(validator -> validator.validate(attribute))
-            .collect(Collectors.toList());
+    List<Serializable> attributeValues = attribute.getValues();
+    List<AttributeValidationReport> resultValidationReportList = new ArrayList<>();
 
-    return generateValidationReports(validationReportList);
+    for (Serializable serializable : attributeValues) {
+
+      List<AttributeValidationReport> validationReportList =
+          validateAttributeSerializable(attribute.getName(), serializable);
+
+      if (validationReportList.size() == validators.size()) {
+        resultValidationReportList.addAll(validationReportList);
+      }
+    }
+
+    return generateValidationReports(resultValidationReportList);
+  }
+
+  private List<AttributeValidationReport> validateAttributeSerializable(
+      String attributeName, Serializable serializable) {
+    List<AttributeValidationReport> validationReportList = new ArrayList<>();
+    for (AttributeValidator attributeValidator : validators) {
+      Attribute newAttribute = new AttributeImpl(attributeName, serializable);
+      Optional<AttributeValidationReport> attributeValidationReport =
+          attributeValidator.validate(newAttribute);
+      if (attributeValidationReport.isPresent()) {
+        validationReportList.add(attributeValidationReport.get());
+      }
+    }
+    return validationReportList;
   }
 
   private Optional<AttributeValidationReport> generateValidationReports(
-      List<Optional<AttributeValidationReport>> validationReportList) {
+      List<AttributeValidationReport> validationReportList) {
 
-    AttributeValidationReportImpl result = new AttributeValidationReportImpl();
+    if (CollectionUtils.isEmpty(validationReportList)) {
+      return Optional.empty();
+    }
 
-    for (Optional<AttributeValidationReport> attributeValidationReportOptional :
-        validationReportList) {
+    AttributeValidationReportImpl result = new AttributeValidationReportImpl();
 
-      if (attributeValidationReportOptional.isPresent()) {
-        AttributeValidationReport attributeValidationReport =
-            attributeValidationReportOptional.get();
+    for (AttributeValidationReport attributeValidationReport : validationReportList) {
 
-        Set<ValidationViolation> validationViolations =
-            attributeValidationReport.getAttributeValidationViolations();
-        Set<String> suggestedValues = attributeValidationReport.getSuggestedValues();
+      Set<ValidationViolation> validationViolations =
+          attributeValidationReport.getAttributeValidationViolations();
+      Set<String> suggestedValues = attributeValidationReport.getSuggestedValues();
 
-        if (CollectionUtils.isEmpty(validationViolations)) {
-          return Optional.empty();
-        }
+      if (CollectionUtils.isEmpty(validationViolations)) {
+        return Optional.empty();
```

Comment:
Thinking through this logic and I'm not sure whether this is the right thing to do. This will return an empty report for the attribute if any one of these individual value reports passes (by returning no violations). Now that we're validating each value individually, there could be other reports for failed values later in the list, right?

Reply:
Looking at this as well; I agree this return should now be removed.  I think the unit tests cover the case where one attribute fails in the middle of the list.